File: lucybell.client/src/app/Models/Categoria.ts
import { SubCategoria } from "./SubCategoria";

export interface Categoria{
  id : number;
  nombre: string;
  urlImagen: string;
  subCategorias: SubCategoria[];
}
export interface CategoriaABM {
  id: number;
  nombre: string;
  urlImagen: string;
}

export interface CategoriaCreacionDTO {
  nombre: string;
}

export interface CategoriaGetSubCategorias {
  id: number;
  nombre: string;
  urlImagen: string;
  isExpanded: boolean;
  subCategorias: SubCategoria[];
}

File: LucyBell.Server/Entidades/Categoria.cs
using System.ComponentModel.DataAnnotations;

namespace LucyBell.Server.Entidades
{
	public class Categoria
	{
		public int Id { get; set; }
		[Required(ErrorMessage = "El campo {0} es requerido")]
		[StringLength(maximumLength: 120, ErrorMessage = "El campo {0} no debe tener más de {1} carácteres")]
		public string Nombre { get; set; }

        public string? UrlImagen { get; set; }
        public List<Producto> Productos { get; set; }
		public List<SubCategoria> SubCategorias { get; set; }
	}
}
